channel/state/outcome/exit.go

```diff
@@ -190,3 +191,42 @@ func (e Exit) Affords(
 	return true
 
 }
+
+// DivertToGuarantee returns a new Exit, identical to the receiver but with
+// (for each asset of the Exit)
+// the leftDestination's amount reduced by leftFunds[asset],
+// the rightDestination's amount reduced by rightAmount[asset],
+// and a Guarantee appended for the guaranteeDestination.
+// Where an asset is missing from leftFunds or rightFunds, it is treated as if the corresponding amount is zero.
+func (e Exit) DivertToGuarantee(
+	leftDestination types.Destination,
+	rightDestination types.Destination,
+	leftFunds types.Funds,
+	rightFunds types.Funds,
+	guaranteeDestination types.Destination,
+) (Exit, error) {
+
+	f := e.Clone()
+
+	for i, sae := range f {
+		asset := sae.Asset
+
+		leftAmount, leftOk := leftFunds.Clone()[asset]
```

Comment:
nit but we could probably clone the `leftFunds` and `rightFunds` once instead of cloning it again for every asset.

Reply:
[0ec03ba](https://github.com/statechannels/go-nitro/pull/262/commits/0ec03ba128e5731c4b5a4dba03247e47a6cb5eb8)